pkg/acsengine/transform/transform.go

```diff
@@ -114,7 +114,8 @@ func (t *Transformer) NormalizeForK8sVMASScalingUp(logger *logrus.Entry, templat
 
 		resourceType, ok := resourceMap[typeFieldName].(string)
 		if ok && resourceType == nsgResourceType {
-			if nsgIndex != -1 {
+			resourceName := resourceMap[nameFieldName].(string)
+			if nsgIndex != -1 && !strings.Contains(resourceName, "variables('jumpboxNetworkSecurityGroupName')") {
```

Comment:
Thinking about this a bit more, I think the `&& !strings.Contains(resourceName, "variables('jumpboxNetworkSecurityGroupName')"` should be on line 116. If the nsg belongs to the jumpbox, we don't want to set the nsgIndex to its index. Otherwise, if the non-jumpbox nsg comes later in the for loop, this will fail (the nsgIndex won't be -1 and it won't belong to the jumpbox).

Reply:
You are correct. I hadn't followed why nsgIndex was set, but get it now. Have made the change.